src/java/org/jivesoftware/openfire/roster/RosterManager.java

```diff
@@ -643,7 +650,10 @@ private void groupUserAdded(Group group, Collection<JID> users, JID addedUser) {
         // Get the roster of the added user.
         Roster addedUserRoster = null;
         if (server.isLocal(addedUser)) {
-            addedUserRoster = rosterCache.get(addedUser.getNode());
+            try {
+                addedUserRoster = getRoster(addedUser.getNode());
+            } catch (UserNotFoundException e) {
+            }
```

Comment:
Please don't catch exceptions without doing anything with them. Even if these exceptions cannot happen, chances are that some future code change might (by accident) cause them to trigger. At the very least log an exception, but if the exception really is "impossible", apply the fail-fast strategy: rethrow it, or wrap it in a runtime exception that's thrown.